src/main.cpp

```diff
@@ -2629,12 +2629,14 @@ int SamApp::RevisionNumber()
 #else
 	wxString path = SamApp::GetRuntimePath() + "/patches/patch_lnx.txt";
 #endif
-	if ( FILE *fp = fopen( (const char*)path.c_str(), "r" ) )
+	std::ifstream fp;
```

Comment:
Changes would have to be tested with Version Upgrade script and with web updated